tests/unit/lib/SGSparseMatrix_unittest.cc

```diff
@@ -23,22 +23,22 @@ using namespace shogun;
 #include <shogun/mathematics/eigen3.h>
 using namespace Eigen;
 
-//function to create sparse matrix
-template<typename T>
-SGSparseMatrix<T> GenerateSparseMatrix(const int32_t numberOfFeatures,
-                                       const int32_t numberOfVectors,
-                                       const float64_t sparseLevel)
+//function can generate both sparse and dense matrix accroding to sparse convention
+template<class MatrixType>
+void GenerateMatrixAccrodingToSparseConvention(const float64_t &sparseLevel,
+                                               const int32_t   &numberOfFeatures,
+                                               const int32_t   &numberOfVectors,
+                                               const int32_t   &randSeed,
```

Comment:
I use const & to denote the variable is not changed inside the function, * to denoting something is changed. If you prefer not to have the reference here, I'm fine with deleting that.